application/pendingtx/pendingindex/pendingtxindexer_test.go

```diff
@@ -0,0 +1,101 @@
+package pendingindex
+
+import (
+	"context"
+	"testing"
+
+	"github.com/MadBase/MadNet/consensus/db"
+	"github.com/MadBase/MadNet/utils"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func createDb() *db.Database {
+	rawEngineDb, err := utils.OpenBadger(context.Background().Done(), "", true)
+	if err != nil {
+		panic(err)
+	}
+	database := &db.Database{}
+	database.Init(rawEngineDb)
+
+	return database
+}
```

Comment:
nit: Function can be made to be a helper which will provide a better test debugging experience than panicing.
```suggestion
func createDB(t *testing.T) *db.Database {
    t.Helper()
	rawEngineDB, err := utils.OpenBadger(context.Background().Done(), "", true)
	if err != nil {
		t.Fatal(err)
	}
	database := &db.Database{}
	database.Init(rawEngineDb)

	return database
}
```